CI Upstream: dask and distributed dev versions have conflicting pinned dependencies; geopandas dev version incompatible with pandas nightly wheels

# Issue 1: Dask and Distributed Dev Version Dependency Conflict

## Problem
Upstream CI testing against development versions of dask and distributed fails during installation with a dependency resolution error:

```
ERROR: Cannot install dask 2025.11.1.dev19+g2f2d09965 (from git+https://github.com/dask/dask.git) 
and distributed==2025.11.1.dev11+g73e68fdcc because these package versions have conflicting dependencies.
The conflict is caused by:
    The user requested dask 2025.11.1.dev19+g2f2d09965
    distributed 2025.11.1.dev11+g73e68fdcc depends on dask==2025.11.0
```

## Root Cause
Development versions of dask and distributed have drifted in their pinned dependencies. When both are installed from git simultaneously, pip's dependency resolver encounters a conflict because `distributed` pins `dask==2025.11.0` while the dask dev version is `2025.11.1.dev19+`.

## Current Workaround
Installing both packages with `--no-deps` bypasses pip's dependency resolution, allowing both development versions to coexist. This works because their shared dependencies (cloudpickle, fsspec, partd, pyyaml) are compatible.

**Related Issues:**
- https://github.com/dask/community/issues/150
- https://github.com/dask/distributed/pull/5802

## Impact
This doesn't block testing entirely (workaround is in place) but indicates version management between dask and distributed could be tightened for dev releases.

---

# Issue 2: GeoPandas/Pandas Dev Version Incompatibility

## Problem
Upstream CI tests fail when installing geopandas from development git with pandas nightly wheels:

```
FAILED test/grid/geometry/test_geometry.py::test_engine_geodataframe 
FAILED test/io/test_geopandas.py::test_read_shpfile
FAILED test/io/test_geopandas.py::test_read_shpfile_multi
FAILED test/io/test_geopandas.py::test_read_shpfile_5poly
FAILED test/io/test_geopandas.py::test_read_geojson
FAILED test/test_plot.py::test_engine

TypeError: pandas._libs.internals.SetitemMixin.__new__(GeoSeries) is not safe, use GeoSeries.__new__()
```

## Root Cause
GeoPandas dev version attempts to instantiate GeoSeries objects using pandas internal APIs that have changed in the pandas nightly build. The geopandas code is using deprecated pandas internals that are no longer safe in the current pandas development version.

## Impact
6 tests fail across geometry, IO, and plotting modules due to this incompatibility. This affects upstream CI validation of uxarray against the latest pandas development code.

--- a/ci/install-upstream.sh
+++ b/ci/install-upstream.sh
@@ -24,15 +24,17 @@ conda list
 
 # if available install from scientific-python nightly wheels
 # constrain numpy to <=2.3 until numba supports newer versions
+# use stable pandas (not nightly) due to geopandas incompatibility with pandas nightly internals
+# (see: https://github.com/UXARRAY/uxarray/issues/1414)
 python -m pip install \
-    'numpy<=2.3'
+    'numpy<=2.3' \
+    'pandas>=2.0.0'
 
 python -m pip install \
     -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple \
     --no-deps \
     --pre \
     --upgrade \
-    pandas \
     scikit-learn \
     scipy \
     xarray

